src/components/PostsList.tsx

```diff
@@ -40,7 +40,7 @@ export const PostsList: React.FC<Props> = ({
                   'is-light': post.id !== selectedPostId,
                 })}
                 onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
+                  setSelectedPost(post.id === selectedPostId ? null : post);
                 }}
```

Comment:
The onClick handler correctly toggles the selected post. Ensure that the setSelectedPost function is handling null values appropriately, as this will determine if a post is deselected.